core/evalir.ml

```diff
@@ -238,7 +238,10 @@ struct
         [], Types.Not_typed)) in
     function
     | `FunctionPtr (f, fvs), ps ->
-      let (_finfo, (xs, body), z, _location) = find_fun f in
+      let (_finfo, (xs, body), z, _location) =
+           try find_fun f
+           with NotFound s ->
+             raise (NotFound (s^" attempting to look up the function name")) in
```

Comment:
Sorry, I think what I had in mind was to raise an Evalir.internal_error here, with a more informative error message than "NotFound", not to change the payload data of NotFound.  

Have a look at the `run_program` function which catches `NotFound` exceptions raised during program execution.  I think we should do something similar here.  (However, the bug illustrated in #967 doesn't seem to be raising a NotFound exception, rather it is finding something but the z and fvs values don't match the first two cases so `assert false` is being raised.  Perhaps the `assert false` should also be turned into a more informative error message.

Reply:
To @jamescheney: Is this more like what you have in your mind? I'll update the previous PR with this. 

try find_fun f
with NotFound s ->
   raise (internal_error s) 

An interesting point is this. 

NotFound exception related with this issue #967 is never propagated to the `with` clause in run_program.
The reason is that `perform_request` (in `webif.ml`) and `callback` (in `webserer.ml`) directly call `Eval.apply` without going through `run_program`.

So the exception does not have any information (like a file name) that `internal_error` appends to it. I think, as a result, it has been a bit hard to find out where this exception is raised.